json/pickler/src/main/scala/sttp/tapir/json/macros.scala

```diff
@@ -0,0 +1,91 @@
+package sttp.tapir.json.macros
+
+import _root_.upickle.implicits.*
+import _root_.upickle.implicits.{macros => uMacros}
+import sttp.tapir.SchemaType
+import sttp.tapir.SchemaType.SProduct
+
+import scala.quoted.*
+
+import compiletime.*
+
+type IsInt[A <: Int] = A
+
+inline def writeSnippets[R, T](
```

Comment:
maybe these all should be package-private? it would also be easier to read, knowing what is the user-facing interface, and what is implementation